src/index.js

```diff
@@ -1,42 +1,44 @@
+/* eslint-disable no-use-before-define */
 import './style.css';
-import { displayBox } from './check.js';
-
-const allTasks = document.querySelector('.task-list');
-
-const listArray = [
-  {
-    description: 'Complete To-Do list',
-    completed: false,
-    index: 0,
-  },
-  {
-    description: 'Take a walk',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'Watch a movie',
-    completed: false,
-    index: 2,
-  },
-];
-
-const showTasks = (tasks) => `
-<li data-id="${tasks.index}">
-<input data-id="${tasks.index}" type="checkbox" name="" class="check-list">
-<label class="list" for="">${tasks.description}<i class="fas fa-ellipsis-v"></i></label>
-</li>
-`;
-allTasks.innerHTML = listArray.map((tasks) => showTasks(tasks)).join('');
-
-displayBox();
-
-// local Storage
-
-const listArraySerialised = JSON.stringify(listArray);
-
-localStorage.setItem('listArray', listArraySerialised);
-
-const listArrayDeserialised = JSON.parse(localStorage.getItem('listArray'));
-
-listArrayDeserialised();
+import { getTodos } from './task.js';
+
+const inputText = document.querySelector('.add');
+const listTask = document.querySelector('.add-task');
+const clear = document.querySelector('.clear-all');
+const todos = JSON.parse(localStorage.getItem('todos')) || [];
+
+const showTasks = (e) => {
+  e.preventDefault();
+  saveLocalTodos({ description: inputText.value, completed: false });
+  getTodos();
+  inputText.value = '';
+};
+listTask.addEventListener('submit', showTasks);
+
+function saveLocalTodos({ description, completed }) {
+  let nextIndex = todos.sort((a, b) => b.index - a.index)[0] || 0;
+  nextIndex = typeof nextIndex === 'object' ? nextIndex.index : 0;
+  nextIndex += 1;
+
+  todos.push({ index: nextIndex, description, completed });
+  localStorage.setItem('todos', JSON.stringify(todos));
+  console.log(todos);
+}
+
+function deleteItem(index) {
```

Comment:
- [ ] It would be better if we implement the functionality where deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. This functionality is currently missing

Reply:
This issue is resolved. :+1: